server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeBuilder.java

```diff
@@ -47,6 +48,13 @@ protected AbstractRangeBuilder(String name, InternalRange.Factory<?, ?> rangeFac
         this.rangeFactory = rangeFactory;
     }
 
+    protected AbstractRangeBuilder(AbstractRangeBuilder<AB, R> clone) {
+        super(clone);
+        this.rangeFactory = clone.rangeFactory;
+        this.ranges = clone.ranges;
```

Comment:
I think we might want to copy the list here rather than share it between the old and new copies? Otherwise if the builder is cloned and then `addRange` is called on the copy it will also change the original?